From 1ea2d3de554fb618184499113251a48f382edd6e Mon Sep 17 00:00:00 2001
From: Dave Sugar <dsugar100@gmail.com>
Date: Tue, 18 Nov 2025 10:34:56 -0500
Subject: [PATCH 15/17] firewalld: Allow firewall-cmd to be called from systemd

RedHat has added 'ExecStartPost=/usr/bin/firewall-cmd --state' in the firewalld.service file.

node=localhost type=USER_AVC msg=audit(1763093230.354:10021): pid=1156 uid=81 auid=4294967295 ses=4294967295 subj=system_u:system_r:system_dbusd_t:s0 msg='avc:  denied  { send_msg } for  scontext=system_u:system_r:initrc_t:s0 tcontext=system_u:system_r:firewalld_t:s0 tclass=dbus permissive=1 exe="/usr/bin/dbus-broker" sauid=81 hostname=? addr=? terminal=?' UID="dbus" AUID="unset" SAUID="dbus"

Signed-off-by: Dave Sugar <dsugar100@gmail.com>
---
 policy/modules/system/init.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/system/init.te b/policy/modules/system/init.te
index 0772961ab..303236c7c 100644
--- a/policy/modules/system/init.te
+++ b/policy/modules/system/init.te
@@ -1315,6 +1315,10 @@ optional_policy(`
 
 	init_dbus_chat(initrc_t)
 
+	optional_policy(`
+		firewalld_dbus_chat(initrc_t)
+	')
+
 	optional_policy(`
 		networkmanager_dbus_chat(initrc_t)
 	')
-- 
2.47.3

